lib/components/user/saved-trip-screen.js

```diff
@@ -0,0 +1,265 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { withLoginRequired } from 'use-auth0-hooks'
+
+import { addTrip, deleteTrip, getTrips, updateTrip } from '../../util/middleware'
+import { routeTo } from '../../actions/ui'
+import DesktopNav from '../app/desktop-nav'
+import { arrayToDayFields, WEEKDAYS } from '../../util/monitored-trip'
+import { getActiveItineraries, getActiveSearch } from '../../util/state'
+import AwaitingScreen from './awaiting-screen'
+import SavedTripEditor from './saved-trip-editor'
+import SavedTripWizard from './saved-trip-wizard'
+import TripBasicsPane from './trip-basics-pane'
+import TripNotificationsPane from './trip-notifications-pane'
+import TripSummaryPane from './trip-summary-pane'
+import withLoggedInUserSupport from './with-logged-in-user-support'
+
+/**
+ * Initializes a monitored trip object from the given query.
+ */
+function getNewMonitoredTrip (loggedInUser, queryParams, itinerary) {
+  return {
+    ...arrayToDayFields(WEEKDAYS),
+    excludeFederalHolidays: true,
+    isActive: true,
+    itinerary,
+    leadTimeInMinutes: 30,
+    queryParams,
+    tripName: '',
+    userId: loggedInUser.id // must provide to API.
+  }
+}
+
+function hasMaxTripCount (trips) {
+  return trips && trips.length >= 5
+}
+
+/**
+ * This screen handles saving a trip from an OTP query for the logged-in user.
+ */
+class SavedTripScreen extends Component {
+  static propTypes = {
+    originalUrl: PropTypes.string
+  }
+
+  static defaultProps = {
+    originalUrl: '/'
+  }
+
+  constructor (props) {
+    super(props)
+
+    const { itinerary, loggedInUser, queryParams, wizard } = props
+    const thisMonitoredTrip = wizard ? getNewMonitoredTrip(loggedInUser, queryParams, itinerary) : null
```

Comment:
Is this wizard prop the same as `isCreating`? If so, please change the prop name.